packages/react/src/ReactChildren.js

```diff
@@ -116,30 +122,42 @@ function traverseAllChildrenImpl(
 ) {
   var type = typeof children;
 
-  if (type === 'undefined' || type === 'boolean') {
-    // All of the above are perceived as null.
-    children = null;
-  }
-
-  if (
-    children === null ||
-    type === 'string' ||
-    type === 'number' ||
-    // The following is inlined from ReactElement. This means we can optimize
-    // some checks. React Fiber also inlines this logic for similar purposes.
-    (type === 'object' && children.$$typeof === REACT_ELEMENT_TYPE) ||
-    (type === 'object' && children.$$typeof === REACT_PORTAL_TYPE)
-  ) {
+  const invokeCallback = () => {
```

Comment:
This is a very hot path. We don't want to create closures here. It's fine to duplicate code instead if necessary.